### PR TITLE
PIL engine: convert tif to png before processing

### PR DESCRIPTION
**Pull request by kkopachev**
This is derived from https://github.com/thumbor/thumbor/pull/714
I decided to put conversion directly into PIL engine, since there might be some other engines (graphicsmagic), which does support tiffs natively.
+ added some tests

### CODE REVIEW

**Comment by cerber717** [comment]
👍

**Comment by guilhermef** [comment]
:+1:  @heynemann what do you think ?

**Comment by kkopachev** [comment]
I amended commit (no changes) to force-restart travis.

**Comment by guilhermef** [comment]
@kkopachev It will be a nice feature, to support tiff, we just need these tests to pass

**Comment by kkopachev** [comment]
@guilhermef Yeah, trying to figure out what's going on there. Passes on my local, though. Looks like cv library version incompatibility. Trying on ubuntu precise now.

**Comment by kkopachev** [comment]
@guilhermef made another commit with a fallback to cv. Looks like there is a bug in one of older python-opencv bindings http://stackoverflow.com/a/24030332

File: tests/engines/test_pil.py [hunk 1]
```diff
@@ -24,6 +24,11 @@
 except ImportError:
     METADATA_AVAILABLE = False
 
+try:
+    from unittest import mock  # Python 3.3 +
```
Comment (guilhermef) [inline tests/engines/test_pil.py]: @kkopachev We don't need python 3 support right now, I think it would just inflate our code base.

**Comment by guilhermef** [comment]
@rfloriano @heynemann It looks good to me

**Comment by kkopachev** [comment]
Any news here?

**Comment by guilhermef** [comment]
@rfloriano @heynemann can I merge ?

**Comment by heynemann** [comment]
thank you very much @kkopachev for your contribution!!!

**Comment by rfloriano** [comment]
Thanks @kkopachev :)

**Pull request merged**